src/task.c

```diff
@@ -359,8 +359,9 @@ task_register_thread(task_system *ts,
                    thread_tid);
 
    thread_tid = task_allocate_threadid(ts);
+   // Unavailable threads is a temporary state that could go away.
    if (thread_tid == INVALID_TID) {
-      return STATUS_NO_SPACE;
+      return STATUS_BUSY;
```

Comment:
This change and L258 above, make the return status to be consistent. (This is how I had it with the previous commit of PR #469 @ SHA 5e8ec1b). The most recent change to `task.c` has introduced this inconsistency which I am reverting back to the way it was previously.

I think this is a better status as seen in the output from this unit-test:

```
TEST 5/6 task_system:test_use_all_but_one_threads_for_bg_threads
Verbose mode on.  This test exercises an error case, so on sucess it will print a message that appears to be an error.
Splinter task system created 1 background thread of type 'TASK_TYPE_MEMTABLE'.
Splinter task system created 61 background threads of type 'TASK_TYPE_NORMAL'.
Cannot create a new thread as the limit on concurrent threads, 64, will be exceeded.
Could not create a thread: Resource temporarily unavailable
```

See the bit above about `Resource temporarily unavailable`.

That comes out of `STATUS_BUSY`, which seems to be a more appropriate error return status in this case of (what usually could be) a transient error.